### PR TITLE
Add TensorToFlow conversion pattern for tensor.insert op

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**
Fixes https://github.com/iree-org/iree/issues/11515

Signed-Off By: Vivek Khandelwal<vivek@nod-labs.com>

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/test/insert.mlir [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// RUN: iree-opt --allow-unregistered-dialect --split-input-file --iree-flow-convert-to-flow %s | FileCheck %s
```
Comment (ScottTodd) [inline compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/test/insert.mlir:1]: nit: It doesn't look like `--allow-unregistered-dialect` should be necessary here. Prefer to keep tests focused on just the flags, passes, ops, etc. that are relevant.

Comment (ScottTodd) [reply compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/test/insert.mlir:1]: I just audited our repo and removed a few other instances of that flag that weren't necessary: https://github.com/iree-org/iree/pull/11535

`insert_slice.mlir` in this folder uses that flag for this test case: https://github.com/iree-org/iree/blob/72242333c11ee9dd7c608979da45b92c3a884d6d/compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/test/insert_slice.mlir#L58-L70

(the `test.` ops aren't from any dialect)

Maybe a similar test case would be useful for this new conversion pattern?

**Comment by vivekkhandelwal1** [comment]
Hi @ScottTodd, I have addressed the comments. Could you please review this PR?

File: compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/test/BUILD [hunk 2]
```diff
@@ -22,6 +22,7 @@ iree_lit_test_suite(
             "fill.mlir",
             "from_elements.mlir",
             "insert_slice.mlir",
+            "insert.mlir",
```
Comment (ScottTodd) [inline compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/test/BUILD:25]: Per the lint check: https://github.com/iree-org/iree/actions/runs/3695690035/jobs/6281665462, this change should also be made in the CMakeLists.txt file. You can run the Bazel -> CMake conversion script to do that for you: `./build_tools/bazel_to_cmake/bazel_to_cmake.py`

**Review by ScottTodd — APPROVED** [review]

**Comment by vivekkhandelwal1** [comment]
Hi @ScottTodd, can you please merge it now?

**Comment by hanhanW** [comment]
I enabled auto-merge

**Pull request merged**